conan/tools/cmake/presets.py

```diff
@@ -78,15 +77,19 @@ def _contents(conanfile, toolchain_file, cache_variables, generator):
     return ret
 
 
-def write_cmake_presets(conanfile, toolchain_file, generator):
-    cache_variables = {}
+def write_cmake_presets(conanfile, toolchain_file, generator, cache_variables):
+    cache_variables = cache_variables or {}
     if platform.system() == "Windows" and generator == "MinGW Makefiles":
-        cache_variables["CMAKE_SH"] = "CMAKE_SH-NOTFOUND"
+        if "CMAKE_SH" not in cache_variables:
+            cache_variables["CMAKE_SH"] = "CMAKE_SH-NOTFOUND"
+
         cmake_make_program = conanfile.conf.get("tools.gnu:make_program", default=None)
```

Comment:
Should we put the `cache_variables.get("CMAKE_MAKE_PROGRAM")` as default?

```suggestion
        cmake_make_program = conanfile.conf.get("tools.gnu:make_program", default=cache_variables.get("CMAKE_MAKE_PROGRAM"))
```

Reply:
yeah, good idea, that way we will replace the "\\" if needed.